transitions/generate_patient_id_on_patients.js

```diff
@@ -0,0 +1,11 @@
+var transitionUtils = require('./utils');
+
+module.exports = {
+  filter: function(doc) {
+    return doc.type === 'person' &&
```

Comment:
> > Have we discussed potentially changing the IDs to person-$id

> I have no idea what this means sorry.

I might be conflating, but pretty recently we discussed using a string like `person-$id` where `$id` is something we generate like a short ID, that is used for people's IDs.  So a doc for a person would looks like:

```
{
  "_id": "person-4994821",
  "type": "person",
  ....
}
```

This is a much better approach in my opinion for a few reasons.  1) If we plan to rely on this feature more and more then we should have a valid identifier for people at creation time  2) It's more efficient, and I don't think very difficult to put the ID generation in couchdb/pouchdb